docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+   2. [Run the tests](Testing.md) to ensure they all pass. 
+//Note: should ask team W11-02 the permission for the above part.
+
+## System Overview
+
+The following sections describe the high level overview of our application, Fluffle.
+
+### Technological view
+The Integrated Development Environment, IntelliJ, is used to develop our program. The program is written in Java, 
+and uses Gradle for building and testing purposes. Our source code is mostly original, with some functions imported 
+from the java.util package. The remaining packages and classes which form the structure of our program 
+are independently developed.
+
+## Architecture
+![UML Fluffle class diagram](graphics/diagrams/classDiagram_Fluffle.png)
+<center><i><b>Figure 1: Overview of Fluffle UML diagram</b></i></center>
+
+Fluffle Documentation: This document contains all packages and classes that are used in developing Fluffle. 
+
+### Project overview
+Fluffle is built using IntelliJ and all concepts for the user interfaces, as well as the backend data management of the 
+application, was created by our team. Due to the restrictions of the project, the main file format used for storage is 
+.txt. We opted to save the data in the text files in a user readable format as opposed to the comma separated format 
+as it is easier for users to directly refer to and edit their saved files.
+
+## Writings class family
+### Constitution (member classes)
+WritingList: Represent the objects which are particular lists of Writings to be used in the application.
+Writings: Represent the objects of the writings, created from user’s input and stored in a database as text. 
+This Writings class is also the parent of 2 subclasses which are Poem and Essay.
+Poem: Represents the Writings objects which have type is Poem.
+Essay: Represents the Writings objects which have type is Essay.
+User: Represents the Users registered to the System
+![UML Class diagram for WritingList family](graphics/diagrams/classDiagram_WritingList.png)
+<p align = "center"><i><b>Figure 2: WritingList family UML diagram</b></i></p>
+
+The above class diagram describes the overall architecture of Writings class functionalities and associations within 
+the scope of related classes. By checking “start”, “type” command with checkStartCommand() then checkTypeCommand() 
+methods on that sequence respectively, the user should be able to access the process of creating and saving new writings 
+into the database. During this process, the user has the ability of choosing their preferred type of writings(which are 
+either poem or essay at this stage)
+
+![UML WritingList family sequence diagram](graphics/diagrams/UML_Writing_Sequence_diagram.png)
+<p align = "center"><i><b>Figure 3: General interactions between member classes when generating a new writing</b></i></p>
+
+## Filter words class family
+![UML Filter word class diagram](graphics/diagrams/classDiagram_FilterWords.png)
```

Comment:
![Fluffle](https://user-images.githubusercontent.com/53225908/97660187-c9a9a480-1aab-11eb-8777-8ce87f2fa9e9.JPG)
Would it be clearer if there are more multiplicities or labeling stated?